pulser-core/pulser/json/abstract_repr/deserializer.py

```diff
@@ -218,6 +233,83 @@ def _deserialize_operation(seq: Sequence, op: dict, vars: dict) -> None:
             channel=op["channel"],
             protocol=op["protocol"],
         )
+    elif op["op"] == "enable_eom_mode":
+        seq.enable_eom_mode(
+            channel=op["channel"],
+            amp_on=_deserialize_parameter(op["amp_on"], vars),
+            detuning_on=_deserialize_parameter(op["detuning_on"], vars),
+            optimal_detuning_off=_deserialize_parameter(
+                op["optimal_detuning_off"], vars
+            ),
+        )
+    elif op["op"] == "add_eom_pulse":
+        seq.add_eom_pulse(
+            channel=op["channel"],
+            duration=_deserialize_parameter(op["duration"], vars),
+            phase=_deserialize_parameter(op["phase"], vars),
+            post_phase_shift=_deserialize_parameter(
+                op["post_phase_shift"], vars
+            ),
+            protocol=op["protocol"],
+        )
+    elif op["op"] == "disable_eom_mode":
+        seq.disable_eom_mode(channel=op["channel"])
+
+
+def _deserialize_channel(obj: dict[str, Any]) -> Channel:
+    params: dict[str, Any] = {}
+    channel_cls: Type[Channel]
+    if obj["basis"] == "ground-rydberg":
+        channel_cls = Rydberg
+        params["eom_config"] = None
+        if obj["eom_config"] is not None:
+            data = obj["eom_config"]
+            params["eom_config"] = RydbergEOM(
+                mod_bandwidth=data["mod_bandwidth"],
+                limiting_beam=RydbergBeam[data["limiting_beam"]],
+                max_limiting_amp=data["max_limiting_amp"],
+                intermediate_detuning=data["intermediate_detuning"],
+                controlled_beams=tuple(
+                    RydbergBeam[beam] for beam in data["controlled_beams"]
+                ),
+            )
+    elif obj["basis"] == "digital":
+        channel_cls = Raman
+    elif obj["basis"] == "XY":
+        channel_cls = Microwave
+
+    for param in dataclasses.fields(channel_cls):
+        if param.init and param.name != "eom_config":
+            params[param.name] = obj[param.name]
+    return channel_cls(**params)
+
+
+def _deserialize_layout(layout_obj: dict[str, Any]) -> RegisterLayout:
+    return RegisterLayout(
+        layout_obj["coordinates"],
+        slug=layout_obj["slug"] if "slug" in layout_obj else None,
+    )
+
+
+def _deserialize_device_object(obj: dict[str, Any]) -> Device | VirtualDevice:
+    device_cls: Type[Device] | Type[VirtualDevice] = (
+        VirtualDevice if obj["is_virtual"] else Device
+    )
+    _channels = tuple(
+        (ch["id"], _deserialize_channel(ch)) for ch in obj["channels"]
+    )
+    params: dict[str, Any] = {"_channels": _channels}
+    for param in dataclasses.fields(device_cls):
+        if not param.init or param.name == "_channels":
```

Comment:
What is param.init ?

Reply:
It's an attribute of a dataclass field. Basically, it's a bool that specifies whether the field is in the `__init__` signature or not.

Reply:
Aren't all fields of a dataclass in the __init__ ?

Reply:
By default yes, but you can choose otherwise. Take for example [`BaseDevice.reusable_channels`](https://github.com/pasqal-io/Pulser/blob/654a1d40ac8d1f7734ee50c619686a05e1a7e146/pulser-core/pulser/devices/_device_datacls.py#L80) - it is explicitly not in the init and always False in a subclass that does not specify otherwise (like `Device`). On `VirtualDevice`, it is in the `__init__` because we can actually choose it to be `True` or `False`.